code/modules/xenoarchaeology/tools/xenoarchaeology_labeler.dm

```diff
@@ -0,0 +1,261 @@
+/obj/item/xenoarchaeology_labeler
+	name = "artifact labeler"
+	icon = 'icons/obj/xenoarchaeology/xenoartifact_tech.dmi'
+	icon_state = "labeler"
+	desc = "A tool scientists use to label their alien bombs."
+	throw_speed = 3
+	throw_range = 5
+	w_class = WEIGHT_CLASS_TINY
+	///Checked traits
+	var/list/selected_traits = list()
+	///List of deselected traits
+	var/list/deselected_traits = list()
+	///List of selected traits we'll put on the label
+	var/list/label_traits = list()
+	///List of possible trait filters
+	var/list/trait_filters = list(list("icon" = "eye", "desc" = "Traits that can appear in the material description."),
```

Comment:
a single new line for internal list() will be nice to read. nitpick for this first one.